csob/resources/__init__.py

```diff
@@ -0,0 +1,188 @@
+from typing import Iterable, Tuple, Dict, Optional
+from urllib.parse import urljoin
+
+import requests
+
+from csob.api_response import APIResponse
+from csob.crypto import get_signature, get_url_signature, verify_signature
+from csob.exceptions import HTTP_ERROR_CSOB_EXCEPTIONS, GatewaySignatureInvalid
+from csob.utils import get_dttm
+
+
+class CSOBResource:
+    url: str
+
+    request_signature: Iterable[str]
+    optional_request_signature: Tuple[str] = tuple()
+    response_signature: Iterable[str]
+    optional_response_signature: Tuple[str] = tuple()
+
+    _base_url: str
+    _gateway_key: str
+    merchant_id: str
+    session: requests.Session
+    raise_exception = True
+
+    def __init__(self, base_url: str, merchant_id: str, gateway_key: str,
+                 session: requests.Session = requests.Session(),
+                 raise_exception: bool = True):
+        self._gateway_key = gateway_key
+        self.raise_exception = raise_exception
+        self.merchant_id = merchant_id
+        self._base_url = base_url
+        self.session = session
+
+    def get_base_json(self) -> dict:
+        return {
+            'merchantId': self.merchant_id,
+            'dttm': get_dttm(),
+        }
+
+    def get_url(self):
+        return urljoin(self._base_url, self.url)
+
+    def _construct_signature_str(self, json: Dict) -> str:
+        """
+        From json constructs signature str.
+
+        Args:
+            json: JSON from which to get the signature str.
+
+        Returns:
+            Signature str
+        """
+        signature_list = []
+        for i in [i for i in self.request_signature if (i not in self.optional_request_signature or i in json.keys())]:
+            if type(json[i]) == list:  # Cart list of dicts
+                for j in json[i]:
+                    signature_list.extend([str(k) for k in j.values()])
+            elif json[i] is True:
```

Comment:
json[i] is True máš kvůli typu že?

Reply:
přesně tak když je to `is` tak by nemělo docházet k přechylování na bool